holopy/core/tests/test_vis.py

```diff
@@ -64,6 +65,37 @@ def convert_ndarray_to_xarray(array, extra_dims=None):
     return array
 
 
+class TestSavingImage(unittest.TestCase):
+    def setUp(self):
+        names = ['image0001', 'image0002', 'image0003']
+        self.holograms = [get_example_data(n) for n in names]
+        self.tempdir = tempfile.mkdtemp()
+        fnames = ['%s.png' % n for n in names]
+        self.filenames = [os.path.join(self.tempdir, f) for f in fnames]
+
+    @attr("fast")
+    def test_save_single_image(self):
+        save_plot(self.filenames[0], self.holograms[0])
+        show(self.holograms[0])
+        fname = os.path.join(self.tempdir, 'test.png')
+        plt.savefig(fname)
+        plt.clf()
+        self._compare_two_images_by_fname(self.filenames[0], fname)
+
+    @attr("fast")
+    def test_save_multiple_images(self):
+        save_plot(self.filenames, self.holograms)
```

Comment:
Hey @MatthiasGi , 

When I clone this branch and run this code on my machine, the ```save_plot``` line crashes with an error:
```python
BadImage: Too many dims on ndarray to output properly.
```
The reason is that the loaded images end up as shape [2, 1, N, N] rather than as 3D images. So I'm a little confused as to exactly what you are trying to do here. Does this code run on your machine? If you can help me out a bit understanding what you are trying to do, I can probably help you fix this and merge it in.

Reply:
I believe the problem is that `self.holograms` is a list of xr.DataArray objects. The `hp.core.io.vis.display_image` function called in `save_plot` is designed to take holograms in a variety of input formats and output a standardized xr.DataArray with known dimension order for saving or displaying. A list of xr.DataArrays is not something it knows how to handle and I think it's getting cast to numpy ndarray.

I think the best solution might be to use `xr.concat` or `hp.core.metadata.clean_concat` to construct the test stack instead of packing as a list.

Reply:
Awesome, thanks @barkls. I fixed this in the next 3 commits. If you could quickly check that I am clean_concat'ing the right way, that would be *awesome*.